### PR TITLE
plugin/Zip: Added support for writing Zip64 when needed.

### PR DESCRIPTION
**Pull request by User0755**
plugin/Zip: Added support for writing Zip64 when needed.

### CODE REVIEW

File: uppsrc/plugin/zip/zip.h [hunk 1]
```diff
@@ -125,8 +127,8 @@ class Zip {
 public:
 	Callback WhenError;
 
-	void BeginFile(const char *path, Time tm = GetSysTime(), bool deflate = true);
-	void BeginFile(OutFilterStream& oz, const char *path, Time tm = GetSysTime(), bool deflate = true);
+	void BeginFile(const char *path, Time tm = GetSysTime(), bool deflate = true, bool zip64 = false);
+	void BeginFile(OutFilterStream& oz, const char *path, Time tm = GetSysTime(), bool deflate = true, bool zip64 = false);
 	void Put(const void *data, int size);
```
Comment (mirek-fidler) [inline uppsrc/plugin/zip/zip.h:132]: Good work overall, I am sort of missing Put(... int64 size).

Comment (User0755) [reply uppsrc/plugin/zip/zip.h:132]: OK, I will see to it... probably add Put64(... int64 size).
// Tom

Comment (User0755) [reply uppsrc/plugin/zip/zip.h:132]: * Do not merge! * I have broken it on some zip64 header offset. I will investigate and fix it and let you know when it's OK again.

Comment (User0755) [reply uppsrc/plugin/zip/zip.h:132]: OK, now I have fixed the offset issue when creating large archives with many files and file sizes. Also, I have added Put64(... int64 size). Is this OK, now?
// Tom

**Pull request merged**